playwright/e2e/create-colony.spec.ts

```diff
@@ -0,0 +1,388 @@
+      await expect(
+        page.getByRole('heading', {
+          name: /Creating a new native token or use existing?/i,
+        }),
+      ).toBeVisible();
+
+      await expect(
+        page.getByRole('button', { name: /continue/i }),
+      ).toBeVisible();
+
+      await expect(page.getByRole('button', { name: /back/i })).toBeVisible();
+
+      await expect(page.getByLabel(/Create a new token/i)).toBeVisible();
+
+      await expect(page.getByLabel(/Use an existing token/i)).toBeVisible();
+    });
+
+    test('Should accept New Colony Token data and navigate to new step', async ({
+      page,
+    }) => {
+      const tokenName = 'Test Token Name';
+      const tokenSymbol = 'TTN';
+
+      await page.getByLabel(/Create a new token/i).check();
+      await page.getByRole('button', { name: /continue/i }).click();
+
+      await expect(
+        page.getByRole('heading', { name: /Your Colony's native token/i }),
+      ).toBeVisible();
+
+      await expect(page.getByLabel(/Create a new token/)).toBeChecked();
+
+      await expect(page.getByLabel(/Use an existing token/i)).not.toBeChecked();
+
+      await page.getByLabel(/token name/i).fill(tokenName);
+
+      await page.getByLabel(/token symbol/i).fill(tokenSymbol);
+    });
+
+    test('Token Logo file uploader should work correctly', async ({ page }) => {
+      const validFilePath =
+        'playwright/fixtures/images/jaya-the-beast_400KB.png';
+
+      const invalidImage =
+        'playwright/fixtures/images/cat-image-new-1_3_MB.png';
+
+      // Select New token option
+      await page.getByLabel(/Create a new token/i).check();
+      await page.getByRole('button', { name: /continue/i }).click();
+
+      // Upload valid image
+      const tokenImageInput = page.locator('input[type="file"]');
+      await tokenImageInput.setInputFiles(validFilePath);
+
+      await expect(
+        page.getByRole('button', { name: /change logo/i }),
+      ).toBeVisible();
+
+      const tokenLogo = page.getByRole('img', { name: /avatar of token/i });
+      await expect(tokenLogo).toBeVisible();
+
+      // Removing the image
+      await page.getByRole('button', { name: /remove/i }).click();
+      await expect(tokenLogo).toBeHidden();
+
+      // Uploading invalid image
+      await tokenImageInput.setInputFiles(invalidImage);
+
+      // Validation message appears
+      const message = page.getByText(
+        /File size is too large, it should not exceed 1MB/i,
+      );
+      await expect(message).toBeVisible();
+      await expect(
+        page.getByRole('button', { name: /try again/i }),
+      ).toBeEnabled();
+
+      await page.getByTestId('file-remove').click();
+
+      await expect(message).toBeHidden();
+      await expect(page.getByTestId('file-remove')).toBeHidden();
+      await expect(
+        page.getByRole('button', { name: /try again/i }),
+      ).toBeHidden();
+    });
+
+    test('Should accept existing token and display validation message', async ({
```

Comment:
Can we have a test where we also check invalid tokens. It should check:
- a bad formatted address
- a good formatted address, which is a user address
- a good formatted address, which is a colony address
- a good formatted address, but one that doesn't exist on the chain
- address zero (it's a good formatted address, exists on the chain, but you shouldn't be allowed to add it as a native token)